app/main.py

```diff
@@ -1,44 +1,74 @@
+import asyncio
 import time
+from typing import Awaitable, Any
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> None:
+    await asyncio.gather(*functions)
+
+
+async def main() -> None:
     # create an IOT service
     service = IOTService()
 
-    # create and register a few devices
-    hue_light = HueLightDevice()
-    speaker = SmartSpeakerDevice()
-    toilet = SmartToiletDevice()
-    hue_light_id = service.register_device(hue_light)
-    speaker_id = service.register_device(speaker)
-    toilet_id = service.register_device(toilet)
-
-    # create a few programs
-    wake_up_program = [
-        Message(hue_light_id, MessageType.SWITCH_ON),
-        Message(speaker_id, MessageType.SWITCH_ON),
-        Message(speaker_id, MessageType.PLAY_SONG, "Rick Astley - Never Gonna Give You Up"),
-    ]
+    # register a few devices
+    res = await asyncio.gather(
+        * [
+            service.register_device(device)
+            for device in (
+                HueLightDevice(), SmartSpeakerDevice(), SmartToiletDevice()
+            )
+        ]
+    )
```

Comment:
```suggestion
    hue_light_id, speaker_id, toilet_id = await asyncio.gather(
        * [
            service.register_device(device)
            for device in (
                HueLightDevice(), SmartSpeakerDevice(), SmartToiletDevice()
            )
        ]
    )
```

Unpack and use these variables